vimscript.html.markdown

```diff
@@ -0,0 +1,572 @@
+echo  1 * 2       |" Multiplication
+echo  1 / 2       |" Division
+echo  1 % 2       |" Modulo (remainder)
+```
+
+### Booleans
+See `:h Boolean`
+
+The number 0 is false, every other number is true. Strings are implicitly
+converted to numbers (see below). There are two pre-defined semantic constants.
+
+```vim
+echo  v:true      |" Evaluates to 1 or the string 'v:true'
+echo  v:false     |" Evaluates to 0 or the string 'v:false'
+```
+
+Boolean values can result from comparison of two objects.
+
+```vim
+echo  1 == 2             |" Equality by value
+echo  1 != 2             |" Unequality
+echo  1 >  2             |" Greater than
+echo  1 >= 2             |" Greater than or equal
+echo  1 <  2             |" Smaller than
+echo  1 <= 2             |" Smaller than or equal
+echo  l1 is l2           |" Equality by identity (lists and dictionaries)
+
+" Regular expression
+echo  "hi" =~ "hello"    |" Regular expression match
+echo  "hi" !~ "hello"    |" Regular expression unmatch
+
+" For string comparison
+echo  "Hi" ==? "hi"      |" String equality, case-insensitive
+echo  "Hi" ==# "hi"      |" String equality, case-sensitive
+" and so on for other other inequality comparisons; comparison is done by
+" alphabetical order
+```
+
+Boolean operations are possible.
+
+```vim
+echo  v:true && v:false       |" Logical AND
+echo  v:true || v:false       |" Logical OR
+echo  ! v:true                |" Logical NOT
+echo  v:true ? 'yes' : 'no'   |" Ternary operator
+```
+
+
+### Strings
+See `:h String`
+
+An ordered zero-indexed sequence of bytes. The encoding of text into bytes
+depends on the option `:h 'encoding'`.
+
+```vim
+" Literal constructors
+echo  "Hello world\n"   |" The last two characters stand for newline
+echo  'Hello world\n'   |" The last two characters are literal
+echo  'Let''s go!'      |" Two single quotes become one quote character
+```
+
+Single-quote strings take all characters are literal, except two single quotes,
+which are taken to be a single quote in the string itself. See `:h expr-quote`
+for all possible escape sequences.
+
+```
+" String concatenation
+echo  'Hello ' .  'world'  |" String concatenation
+echo  'Hello ' .. 'world'  |" String concatenation
+
+" String indexing
+echo  'Hello'[0]           |" First byte
+echo  'Hello'[1]           |" Second byte
+echo  'Hello'[-1]          |" Last byte
+echo  'Hello'[-2]          |" Second to last byte
+echo  'Hellö'[4]           |" Returns a byte, not the character 'ö'
+
+" Substrings
+echo  'Hello'[:]           |" Copy of entire string
+echo  'Hello'[1:3]         |" Substring, second to fourth byte
+echo  'Hello'[1:-2]        |" Substring until second to last byte
+echo  'Hello'[1:]          |" Substring with starting index
+echo  'Hello'[:2]          |" Substring with ending index
+```
+
+A negative index is relative to the end of the string. See `:h
+string-functions` for all string-related functions.
```

Comment:
You could point out that, differently from python, the second index in the slice is *included*, so that `hello[:-1]` is unfortunately the same as `hello[:]` and `hello`, and if you want the string *up to but not including the last byte* you should write `hello[:-2]`, for the same reason line 135 could be wrong (it's a substring *until the last byte*, not the second to last byte)

Reply:
I added a little note that the second index is inclusive. It is worth mentioning since in most languages the second index is exclusive, but I agree that comparisons with other languages are out of place here.

Reply:
ok but vimL is the only language I ever saw with this (questionable) characteristic, I think it's important to point it out.